### ISSUE TITLE
[Bug]: Helicone with Azure doesn't work

### ISSUE THREAD

**Issue by saasjesus** [Helicone/helicone#2582]
### What happened?

I want to use Azure as my OpenAi Provider.

I have created a deployment on Azure and received my credentials.

Upon following the [Doc here](https://docs.helicone.ai/integrations/azure/javascript) to connect to Helicone with Azure (NodeJs) -> It just doesn't work.

1. First of all, when I have my config just like in the Doc (without the `apiKey` property in the root). Like so:
```
import OpenAI from "openai"

const openai = new OpenAI({
  baseURL: "https://oai.helicone.ai/openai/deployments/[DeploymentName]",
  defaultHeaders: {
    "Helicone-Auth": `Bearer ${process.env.HELICONE_API_KEY}`,
    "Helicone-OpenAI-API-Base": "https://chatarmin-ai.openai.azure.com",
    "api-key": process.env.AZURE_API_KEY,
    "Helicone-Cache-Enabled": "true",
  },
  defaultQuery: { "api-version": "2024-05-13" },
})
```
This throws me an Error: "The OPENAI_API_KEY environment variable is missing or empty; either provide it, or instantiate the OpenAI client with an apiKey option, like new OpenAI({ apiKey: 'My API Key' })."

2. So when adding the `apiKey` the error doesn't appear, but when I now try to do an embedding like so:
```
await openai.embeddings.create({
      input: [text],
      model: "text-embedding-3-small",
    })

```

I just get a 404 Error, saying "Resource not found".

So basically the Azure Doc of Helicone has to be outdated.

I am using "openai": "^4.58.1"

**Comment by devinat1** [comment]
Hi @saasjesus have you tried making a request in this format:
```
curl --request POST \
  --url 'http://oai.helicone.ai/openai/deployments/gpt-4o-mini/chat/completions?api-version=2025-01-01-preview' \
  --header 'Content-Type: application/json' \
  --header 'Helicone-Auth: Bearer <HELICONE_API_KEY>' \
  --header 'Helicone-OpenAI-Api-Base: "https://chatarmin-ai.openai.azure.com", \
  --header 'api-key: <API_KEY>' \
  --data '{
	"messages": [
		{
			"role": "user",
			"content": "hi"
		}
	],
	"max_tokens": 1,
	"temperature": 1,
	"model": "gpt-4o-mini",
	"stream": true
}'
```
